_posts/en/newsletters/2020-04-29-newsletter.md

```diff
@@ -0,0 +1,330 @@
+    spender (such as Alice in Alice→Mallory), it may produce a "payment
+    failed" error that causes the user to initiate a second payment
+    without realizing the first payment hasn't been revoked.  This
+    [indirect attack][corallo send twice] can perhaps be dealt with by warning the user that
+    the payment is stuck---not failed---and that sending additional
+    payments could result in losses.
+
+{% include references.md %}
+{% include linkers/issues.md issues="688,15761,17509,14895" %}
+[bitcoin core 0.20.0]: https://bitcoincore.org/bin/bitcoin-core-0.20.0
+[Hierarchical Deterministic ref]: https://bitcoin.org/en/glossary/hd-protocol
+[multi-wallet]: https://bitcoin.org/en/release/v0.15.0.1#multi-wallet-support
+[lnd 0.10.0-beta]: https://github.com/lightningnetwork/lnd/releases/tag/v0.10.0-beta.rc5
+[c-lightning 0.8.2]: https://github.com/ElementsProject/lightning/releases/tag/v0.8.2rc2
+[news78 anchor outputs]: /en/newsletters/2019/12/28/#anchor-outputs
+[corallo thread bd]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-April/017757.html
+[corallo thread ld]: https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-April/002639.html
+[harding reject]: https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-April/002650.html
+[corallo targeted relay]: https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-April/002652.html
+[harding pay]: https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-April/002664.html
+[zmn ptlcs]: https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-April/002667.html
+[osuntokun reasonable]: https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-April/002641.html
+[corallo mempool not guaranteed]: https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-April/002648.html
+[darosior revault]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2020-April/017793.html
+[news94 bishop vault]: /en/newsletters/2020/04/22/#vaults-prototype
+[cltv_expiry_delta]: https://github.com/lightningnetwork/lightning-rfc/blob/master/02-peer-protocol.md#cltv_expiry_delta-selection
+[cl ced]: https://github.com/ElementsProject/lightning/blob/10f47b41fa3192638442ef04d816380950cc32c9/lightningd/options.c#L630
+[eclair ced]: https://github.com/ACINQ/eclair/blob/19975d3d8128705b92811ce0bc7a3881ecaf99dd/eclair-core/src/main/resources/reference.conf#L61
+[lnd ced]: https://github.com/lightningnetwork/lnd/blob/0cf63ae8981a1041dd7b9f217dcc2158a28247d3/chainregistry.go#L64
+[rl ced]: https://github.com/rust-bitcoin/rust-lightning/blob/12e2a81e1daf635578e1cfdd7de55324ed04bd48/lightning/src/ln/channelmanager.rs#L430
+[simplicity]: https://blockstream.com/simplicity.pdf
+[bitcoin core default ancestor limit]: https://github.com/bitcoin/bitcoin/blob/9fac600ababd8edefbe053a7edcd0e178f069f84/src/validation.h#L56
+[corallo slightly higher]: https://github.com/bitcoinops/bitcoinops.github.io/pull/394#discussion_r416014263
```

Comment:
I dont like this link (or the one below) as it doesn’t seem to add much. Im not strong on it.

Reply:
I'd prefer to keep these links.  Most of the assertions made in this section are either LN basics or are covered in the linked thread, so I don't feel a need to cite every little fact.  But the claims that 'the problem affects non-routing nodes' and that 'higher base commitment transaction fees reduce the need for package relay in the short term' are not covered in the thread, so I think it's important to link to my sources even if they don't add much additional information.

I know you said that you don't feel strongly about this, but if you change your mind or if anyone else feels strongly about the point, please feel free to make a last-minute edit changing the links to HTML comments.  That way I can feel like I've cited my sources but normal readers won't see the links.

Reply:
Fine to leave them!